pkg/imgutil/dockerconfigresolver/credentialsstore_test.go

```diff
@@ -177,6 +178,12 @@ func TestBrokenCredentialsStore(t *testing.T) {
 		for _, tc := range testCases {
 			t.Run(tc.description, func(t *testing.T) {
 				directory := tc.setup()
+				// See https://github.com/containerd/nerdctl/issues/3413
+				oldpwd, err := os.Getwd()
```

Comment:
> this test is not parallelized.

Yes, and it should stay that way.

> rollback dir at the end of the test

Geez. My bad, and absolutely yes. Will fix.

Reply:
`oldpwd` should be outside the loop otherwise `os.Getwd()` of the second iteration is equal to  `directory`because `t.Cleanup` is  executed once by the end of the test.